### PR TITLE
Feature/8852 smooch bot enhancements archived column

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

File: app/controllers/test_controller.rb [hunk 1]
```diff
@@ -158,7 +158,7 @@ def new_bot
 
   def archive_project
     p = Project.find(params[:project_id])
-    p.archived = true
+    p.archived = 1
```
Comment (caiosba) [inline app/controllers/test_controller.rb:161]: So you changed `archived` from `Team` and `Project` to be integers too?

File: lib/check_archived_flags.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module CheckArchivedFlags
+  NONE = 0
+  TRASHED = 1
+  UNCONFIRMED = 2
+end
```
Comment (caiosba) [inline lib/check_archived_flags.rb:5]: Sawy, the best way to implement that is as a standalone module (like https://github.com/meedan/check-api/blob/develop/lib/error_codes.rb), not a module to be included in all classes.

File: lib/tasks/migrate/20201124120741_convert_archived_flag_to_integer.rake [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+          print '.'
+          ids = versions.map(&:id)
+          Version.from_partition(team.id).where(id: ids).update_all(object_changes: "{\"archived\":[0,1]}")
+        end
+      end
+      minutes = ((Time.now.to_i - started) / 60).to_i
+      puts "[#{Time.now}] Done in #{minutes} minutes."
+    end
+  end
+end
```
Comment (caiosba) [inline lib/tasks/migrate/20201124120741_convert_archived_flag_to_integer.rake:23]: Did you try that with a live dump already?

File: test/controllers/elastic_search_3_test.rb [hunk 4]
```diff
@@ -116,7 +116,7 @@ def setup
     result = CheckSearch.new({ show: ['claims', 'links', 'images'] }.to_json)
     assert_equal 3, result.medias.count
     # filter by archived
-    pm.archived = true
+    pm.archived = 1
```
Comment (caiosba) [inline test/controllers/elastic_search_3_test.rb:119]: Why not use the constants in the test too?

File: test/controllers/graphql_controller_2_test.rb [hunk 5]
```diff
@@ -312,7 +312,7 @@ def setup
     pm = create_project_media project: p
     pm1 = create_project_media project: p, user: u
     create_relationship source_id: pm.id, target_id: pm1.id
-    pm1.archived = true
+    pm1.archived = 1
```
Comment (caiosba) [inline test/controllers/graphql_controller_2_test.rb:315]: Why not use the constants in the test too?

File: test/controllers/graphql_controller_2_test.rb [hunk 6]
```diff
@@ -1157,7 +1157,7 @@ def setup
     team = create_team
     create_team_user team: team, user: u, role: 'owner'
     p = create_project team: team
-    create_project_media archived: true, project: p
+    create_project_media archived: 1, project: p
```
Comment (caiosba) [inline test/controllers/graphql_controller_2_test.rb:1160]: Why not use the constants in the test too?

File: test/controllers/graphql_controller_3_test.rb [hunk 7]
```diff
@@ -61,7 +61,7 @@ def setup
     end
     create_project_media project: p, user: u, disable_es_callbacks: false
     pm = create_project_media project: p, disable_es_callbacks: false
-    pm.archived = true
+    pm.archived = 1
```
Comment (caiosba) [inline test/controllers/graphql_controller_3_test.rb:64]: Why not use the constants in the test too?

File: test/controllers/graphql_controller_3_test.rb [hunk 8]
```diff
@@ -188,7 +188,7 @@ def setup
     pm1b = create_project_media project: p1b, disable_es_callbacks: false ; sleep 1
     pm1b.disable_es_callbacks = false ; pm1b.updated_at = Time.now ; pm1b.save! ; sleep 1
     pm1a.disable_es_callbacks = false ; pm1a.updated_at = Time.now ; pm1a.save! ; sleep 1
-    pm1c = create_project_media project: p1a, disable_es_callbacks: false, archived: true ; sleep 1
+    pm1c = create_project_media project: p1a, disable_es_callbacks: false, archived: 1 ; sleep 1
```
Comment (caiosba) [inline test/controllers/graphql_controller_3_test.rb:191]: Why not use the constants in the test too?

**Review by caiosba — CHANGES_REQUESTED** [review]

File: lib/tasks/migrate/20201124120741_convert_archived_flag_to_integer.rake (hunk 3 above)
Comment (melsawy) [reply lib/tasks/migrate/20201124120741_convert_archived_flag_to_integer.rake:23]: Yes Caio tested locally with a live dump

File: app/controllers/test_controller.rb (hunk 1 above)
Comment (melsawy) [reply app/controllers/test_controller.rb:161]: yes and for `Source` too

File: test/controllers/elastic_search_3_test.rb (hunk 4 above)
Comment (melsawy) [reply test/controllers/elastic_search_3_test.rb]: Fixed by using constants

File: test/controllers/graphql_controller_2_test.rb (hunk 5 above)
Comment (melsawy) [reply test/controllers/graphql_controller_2_test.rb]: Fixed by using constants

File: test/controllers/graphql_controller_2_test.rb (hunk 6 above)
Comment (melsawy) [reply test/controllers/graphql_controller_2_test.rb]: Fixed by using constants

File: test/controllers/graphql_controller_3_test.rb (hunk 7 above)
Comment (melsawy) [reply test/controllers/graphql_controller_3_test.rb]: Fixed by using constants

File: test/controllers/graphql_controller_3_test.rb (hunk 8 above)
Comment (melsawy) [reply test/controllers/graphql_controller_3_test.rb]: Fixed by using constants

File: lib/check_archived_flags.rb (hunk 2 above)
Comment (melsawy) [reply lib/check_archived_flags.rb:7]: Done

**Pull request merged**